Sources/Shared/Core/AuthenticatingListeningDelegate.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  AuthenticatingListeningDelegate.swift
+//  VimeoNetworking
+//
+//  Created by Rogerio de Paula Assis on 10/16/19.
+//  Copyright © 2019 Vimeo. All rights reserved.
+//
+
+import Foundation
+
+/// A type that listens to and responds to authentication status changes
+public protocol AuthenticationListeningDelegate {
```

Comment:
`AuthenticationListening` or `AuthenticationListener`? or does delegate help infer something else

Reply:
Yes good call! At first I wanted to separate SessionManaging and AuthenticationListening as separate types because it felt they are responsible for very different things however after looking at this again I think it makes more sense to simplify this to one type - `SessionManaging` that inherits from `AuthenticationListener`.